app/js/streaming/TimeSyncController.js

```diff
@@ -0,0 +1,304 @@
+                MILLISECONDS_IN_SECONDS = 1000,
+                datetimeRegex = /^([0-9]{4})-([0-9]{2})-([0-9]{2})T([0-9]{2}):([0-9]{2})(?::([0-9]*)(\.[0-9]*)?)?(?:([+\-])([0-9]{2})([0-9]{2}))?/,
+                match = datetimeRegex.exec(xsdatetimeStr),
+                utcDate,
+                timezoneOffset;
+
+            // If the string does not contain a timezone offset different browsers can interpret it either
+            // as UTC or as a local time so we have to parse the string manually to normalize the given date value for
+            // all browsers
+            utcDate = Date.UTC(
+                parseInt(match[1], 10),
+                parseInt(match[2], 10) - 1, // months start from zero
+                parseInt(match[3], 10),
+                parseInt(match[4], 10),
+                parseInt(match[5], 10),
+                (match[6] && (parseInt(match[6], 10) || 0)),
+                (match[7] && parseFloat(match[7]) * MILLISECONDS_IN_SECONDS) || 0
+            );
+            // If the date has timezone offset take it into account as well
+            if (match[9] && match[10]) {
+                timezoneOffset = parseInt(match[9], 10) * MINUTES_IN_HOUR + parseInt(match[10], 10);
+                utcDate += (match[8] === '+' ? -1 : +1) * timezoneOffset * SECONDS_IN_MIN * MILLISECONDS_IN_SECONDS;
+            }
+
+            return new Date(utcDate).getTime();
+        },
+
+        // try to use the built in parser, since xsdate is a contrained ISO8601
+        // which is supported natively by Date.parse. if that fails, try a
+        // regex-based version used elsewhere in this application.
+        xsdatetimeDecoder = function (xsdatetimeStr) {
+            var parsedDate = Date.parse(xsdatetimeStr);
+
+            if (isNaN(parsedDate)) {
+                parsedDate = alternateXsdatetimeDecoder(xsdatetimeStr);
+            }
+
+            return parsedDate;
+        },
+
+        // takes ISO 8601 timestamp and returns milliseconds since UNIX epoch
+        iso8601Decoder = function (isoStr) {
+            return Date.parse(isoStr);
+        },
+
+        // takes RFC 1123 timestamp (which is same as ISO8601) and returns
+        // milliseconds since UNIX epoch
+        rfc1123Decoder = function (dateStr) {
+            return Date.parse(dateStr);
+        },
+
+        notSupportedHandler = function () {
+            return new Promise(function (resolve, reject) {
+                reject(new Error("Not supported"));
+            });
+        },
+
+        directHandler = function (xsdatetimeStr) {
+            return new Promise(function (resolve, reject) {
+                var time = xsdatetimeDecoder(xsdatetimeStr);
+
+                if (isNaN(time)) {
+                    reject(time);
+                }
+
+                resolve(time);
+            });
+        },
+
+        httpHandler = function (decoder, url, isHeadRequest) {
+            return new Promise(function (resolve, reject) {
```

Comment:
Fair point - I did not realise IE11 did not support Promise.

Promises here were used as sugar to avoid having to explicitly pass callbacks to the XHR, so can be removed easily. Looking at it now, that may make it easier to support multiple entries in the `value`.